### ISSUE TITLE
Validate Downloaded images

### ISSUE THREAD

**Issue by TehBotolSosro** [fmd-project-team/FMD#25]
Hi,

I'm usually download from batoto or mangastream but sometime FMD didn't get few pages/images in a chapter so i had to redownload to get the missing pages.

is there a way for FMD to check if all the images/pages are all downloaded or not and restart the download automatically, maybe like count the total of image/page before download and do a count of total image downloaded if its different then restart the download.

in my case the issue happen quite often (probably 1 chapter have missing page every time i download 4 chapter), also it's hard to check manually when it's only have 1-2 page missing

Example. I download one piece c999
it have 20 image/page, fmd sometime only download 13 of them so i had to redownload c999 so i can get the missing images/pages

also thanks for taking over the project since it's hasn't been updated for a while

**Comment by Von07** [comment]
Glad I found a same issue like this one because I experience this just now from a site http://holymanga.net
so I thought the problem is the website itself but when I tried to check it, the web. is complete while the fmd chapters has lacking pages from the web.
So I hope that this will be fix soon

Thank you for taking over this project ^^

**Comment by TehBotolSosro** [comment]
yeah what's weird is that the task didn't show as failed task but as a completed task, 
i also didn't use too many connection ( 2 downloaded task and 2 files per task at the same time)
and even have the retry option enabled (5 retry and 30 sec timeout), but still sometime FMD only download 1 page in mangastream

it could be because the websites drop the connection it's full or maybe there some protection like in madokami website which sometime give 404 error when you start another download while you still downloading some chapter from there, so the only to resolve is to use one connection or keep retrying when it give 404 error

**Comment by panhartstuff** [comment]
Something more useful would probably a better error handling feature. Maybe something like a queue list of failed downloads you can retry later on.
Also I wanna ask, is there a gap in the filenames (12,13,16,17,etc.) when some downloads failed?

**Comment by TehBotolSosro** [comment]
> 
> 
> Something more useful would probably be a better error handling feature. Maybe something like a queue list of failed downloads (for individual pages) you can retry later on.
> Also I wanna ask, is there a gap in the filenames (12,13,16,17,etc.) when some downloads failed?

yes there's a gap in the filename, (eg, 01, 02,04,09), yeah a better error handling sound good, i tried using charles proxy yesterday to see what went wrong but sadly in that session everything downloaded fine, seem something to do with multiple download

**Comment by TehBotolSosro** [comment]
is there any news regarding this? as it still occasionally happen, and it's hard to check manually especially if you download many chapters at once

**Comment by Tmp341** [comment]
> 
> 
> is there any news regarding this? as it still occasionally happen, and it's hard to check manually especially if you download many chapters at once

This FMD branch is not getting updated regarding coding part (adding features like this etc), so create issue on more active repo [FMD2](https://github.com/dazedcat19/FMD2).

**Comment by TehBotolSosro** [comment]
thank you will post in fmd2
